src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -1648,6 +1662,20 @@ int dhcp_lease_load(sd_dhcp_lease **ret, const char *lease_file) {
                         log_debug_errno(r, "Failed to parse T2 %s, ignoring: %m", t2);
         }
 
+        if (saved_realtime && saved_boottime) {
```

Comment:
Then, if we read BOOTTIME, check the BOOTID, and if it matches the current boot id, then set all timestamps from the boottime.

If BOOTID does not match the current boot id, ignore BOOTTIME, and set timestamp from the realtime.

If both realtime and boottime is not set, maybe better to discard the saved lease, as we do not know if it is still valid.